justfile

```diff
@@ -11,6 +11,65 @@ check-format:
     just --unstable --fmt --check -f justfile
     yarn prettier --no-error-on-unmatched-pattern --check **/*.yml **/*.json **/*.md
 
+# Generates the daemonset manifest for kube-vip.
+[group('codegen')]
+codegen-kube-vip-daemonset:
+    #!/usr/bin/env bash
+    set -euo pipefail
+
+    # renovate: datasource=docker depName=ghcr.io/kube-vip/kube-vip
+    KUBE_VIP_VERSION=v0.8.10
+    DEST=kustomization/components/kube-vip/daemonset.yml
+    SCRATCH=/tmp/daemonset.yml
+
+    # First, generate the manifest with their tooling.
+    docker run --network host --rm ghcr.io/kube-vip/kube-vip:${KUBE_VIP_VERSION} \
+        manifest daemonset \
+            --address=8.8.8.8 \
+            --arp \
+            --controlplane \
+            --inCluster \
+            --leaderElection \
+            --namespace=kube-vip \
+            --services \
+            --servicesElection \
+            --taint \
+        > ${SCRATCH}
+
+    # Update namespace to kube-vip (optional patch is documented).
+    yq -iy '.metadata.namespace = "kube-vip"' ${SCRATCH}
```

Comment:
I like how there is yet one more yaml patch syntax to learn...but this one seems nicer than the k8s ones